### ISSUE TITLE
Add support for PEP695

### ISSUE THREAD

**Issue by Daverball** [snok/flake8-type-checking#172]
This is mostly so we don't forget: [PEP695](https://peps.python.org/pep-0695/) introduced a new syntax for generics with type vars that are defined inline with function/class definitions.

For the most part we probably won't have to do anything, but there is a potential for some false positives in the cases where a type var defined in a generic shadows a type var that is defined/imported inside a type checking block and then incorrectly emits TC200/TC100.

E.g. the following would probably result in a TC200/TC100 for all these cases
```python
if TYPE_CHECKING:
    from .types import T

def foo[T](a: T) -> T: ...   # TC200/TC100 for T

type Foo[T] = T | None   # TC200/TC100 for T

class Bar[T]:
    x: T   # TC200/TC100 for T
```

We would need to keep track of the names of any type parameters stored on `ClassDef`/`FunctionDef`/`TypeAlias` nodes and record them alongside a `lineno`/`col_offset`/`end_lineno` triplet akin to the type checking blocks, so we can ignore unquoted annotations that match one of the type parameters in an enclosing block.

**Comment by sondrelg** [comment]
I had totally missed 695. That's actually so cool
